mdtraj/core/trajectory.py

```diff
@@ -369,69 +369,102 @@ def load(filename_or_filenames, discard_overlapping_frames=False, **kwargs):
+    topkwargs = kwargs.copy()
+    #if top is not given try with one of the trajectory files
+    topkwargs.pop("top", None)
+    topkwargs.pop("atom_indices", None)
+    topkwargs.pop("frame", None)
+    topkwargs.pop("stride", None)
+    topkwargs.pop("start", None)
+    kwargs["top"] = _parse_topology(kwargs.get("top", filename_or_filenames[0]), **topkwargs)
+
+    #get the right loader
     try:
         #loader = _LoaderRegistry[extension][0]
         loader = FormatRegistry.loaders[extension]
     except KeyError:
         raise IOError('Sorry, no loader for filename=%s (extension=%s) '
-                      'was found. I can only load files '
-                      'with extensions in %s' % (filename, extension, FormatRegistry.loaders.keys()))
+                    'was found. I can only load files '
+                    'with extensions in %s' % (
+                        filename_or_filenames[0], extension, FormatRegistry.loaders.keys()))
+
+    if loader.__name__ not in ['load_dtr']:
+            _assert_files_exist(filename_or_filenames)
+    else:
+        _assert_files_or_dirs_exist(filename_or_filenames)
+
 
-    if extension in _TOPOLOGY_EXTS:
+    if extension not in _TOPOLOGY_EXTS:
+        # standard_names is a valid keyword argument only for files containing topologies
+        kwargs.pop('standard_names', None)
+
+    trajectories = []
+    tmp_file = filename_or_filenames.pop(0)
+    try:
         # this is a little hack that makes calling load() more predictable. since
         # most of the loaders take a kwargs "top" except for load_hdf5, (since
         # it saves the topology inside the file), we often end up calling
         # load_hdf5 via this function with the top kwarg specified. but then
         # there would be a signature binding error. it's easier just to ignore
         # it.
-        if 'top' in kwargs:
-            warnings.warn('top= kwarg ignored since file contains topology information')
-            kwargs.pop('top', None)
-    else:
-        # standard_names is a valid keyword argument only for files containing topologies
-        kwargs.pop('standard_names', None)
-
-    if loader.__name__ not in ['load_dtr']:
-        _assert_files_exist(filename_or_filenames)
-    else:
-        _assert_files_or_dirs_exist(filename_or_filenames)
-
-    value = loader(filename, **kwargs)
-    return value
+        #TODO make all the loaders accept a pre parsed topology (top) in order to avoid
+        #this part and have a more consistent interface and a faster load function
+        t = loader(tmp_file, **kwargs)
+        
+    except TypeError as e:
+
+        #Don't want to intercept legit
+        #TypeErrors
+        if "got an unexpected keyword argument 'top'" not in str(e):
+            raise
+
+        kwargs.pop('top', None)
+
+        t = loader(tmp_file, **kwargs)
+
+    #in case only a part of the atoms where selected
+    #I get the right topology to laret share with the
+    #other frames
```

Comment:
Missing spaces after `#` and a typo
```suggestion
    # In case only a part of the atoms were selected
    # I get the right topology to later share with the
    # other frames
```